src/circuit/gadget/sinsemilla/note_commit.rs

```diff
@@ -1590,9 +1581,12 @@ mod tests {
 
                 // Witness g_d
                 let g_d = {
-                    let g_d = self.gd_x.map(|x| {
+                    let g_d = self.gd_x.zip(self.gd_y_lsb).map(|(x, y_lsb)| {
                         // Calculate y = (x^3 + 5).sqrt()
-                        let y = (x.square() * x + pallas::Affine::b()).sqrt().unwrap();
+                        let mut y = (x.square() * x + pallas::Affine::b()).sqrt().unwrap();
+                        if y.is_odd() && y_lsb == pallas::Base::zero() {
```

Comment:
This only covers half of the required sign flips. Easiest way to cover both is with an XOR:
```suggestion
                        if y.is_odd() ^ y_lsb.is_odd() {
```